roles/network_plugin/cilium/templates/cilium/config.yml.j2

```diff
@@ -200,6 +200,7 @@ data:
 {% if cilium_encryption_type == "wireguard" %}
   enable-wireguard: "true"
   enable-wireguard-userspace-fallback: "{{ cilium_wireguard_userspace_fallback }}"
+  enable-l7-proxy: "false"
```

Comment:
According to https://github.com/cilium/cilium/issues/15462 the cilium community seems to consider that is an issue and try to solve it.
It is better to write some `#NOTE:` here to explain this is just a workaround and we will be able to remove this after fixing the issue on the cilium side.